routes/register.js

```diff
@@ -0,0 +1,54 @@
+var express = require('express');
+const Users = require('../models/users.js');
+
+const server = express();
+
+server.use(express.json());
+
+server.post('/register', async(req, res) => {
+    const { username, email, phonenumber, password, confirmPassword } = req.body;
+    // check if there already exists a user with the same email, username and phonenumber
```

Comment:
Instead of performing 3 different "findOne's" on Users, you could perform just one and group your conditions using $or operand, which will return the first entry whether one of the conditions is satisfied. And after that, if there is 1 user found, just check which of the field is equal to your parameters in order to return custom error message.
Something like:

```suggestion
    // check if there already exists a user with the same email, username and phonenumber
    const findUser = await Users.findOne({  
        $or: [
             { username: username },
             { phonenumber: phonenumber },
             { email: email }
        ]
    });
    
    if (findUser.username === username) {
         return res.json({
             msg: "This username has already been used"
         });
     }
     
     if (findUser.email === email) {
         return res.json({
             msg: "This email address has already been used"
         });
     }
     
     if (findUser.phonenumber === phonenumber) {
         return res.json({
             msg: "This phone number has already been used"
         });
     }
```

Reply:
I am getting an error here. If there's no user in the db that satisfies one of the 3 conditions then findUser will be null and I am getting an error at the first if (the one that checks username)
I wish to ask if this suggestion is alright (it works for me when running the app and testing with postman)
```suggestion
    // check if there already exists a user with the same email, username and phonenumber
    const findUser = await Users.findOne({  
        $or: [
             { username: username },
             { phonenumber: phonenumber },
             { email: email }
        ]
    });
    
    if(findUser){
        if(findUser.username === username) {
            return res.json({
                msg: "This username has already been used"
            });
        }
        
        if(findUser.email === email) {
            return res.json({
                msg: "This email address has already been used"
            });
        }
    
        if(findUser.phonenumber === phonenumber) {
            return res.json({
                msg: "This phone number has already been used"
            });
        }
    }
```

Reply:
It was just a suggestion, not a complete solution :)
But for the case when findUser is null you can either wrap all 3 if statements in another if checking if there is an user found or you can just use optional chaining for the findUser, like:

`if ( findUser?.username === username )` 

If findUser is null the optional chaining '?.' will give you an 'undefined' value and will not throw an error, now you will be comparing undefined === username

Reply:
thank you! I implemented and pushed the suggestions (also applied the optional chaining for the 3 if statements used for custom error message)